pkg/externalendpoint/disk_manifest_loader.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalendpoint
+
+import (
+	"bufio"
+	"bytes"
+	"os"
+	"path/filepath"
+
+	"github.com/dapr/dapr/utils"
+
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const yamlSeparator = "\n---"
+
+// manifestLoader loads a specific manifest kind from a folder.
+type DiskManifestLoader[T kubernetesManifest] struct {
```

Comment:
Isn't this very similar to `pkg/components/disk_manifest_loader.go` (which is using a generic too)? Could we use a single manifest loader for components, resiliency policies, and external HTTP endpoints?

Reply:
yes, I can look at refactoring this

Reply:
resolving and will create follow up issues per feedback
https://github.com/dapr/dapr/pull/6227/#discussion_r1164342603